pkg/kademlia/routing_helpers.go

```diff
@@ -290,10 +307,22 @@ func (rt *RoutingTable) getNodeIDsWithinKBucket(bucketID storage.Key) (storage.K
 	left := endpoints[0]
 	right := endpoints[1]
 	var nodeIDs storage.Keys
-	allNodeIDs, err := rt.nodeBucketDB.List(nil, 0)
+
+	allNodeItems, _, err := rt.nodeBucketDB.List(storage.ListOptions{
+		Start: nil,
+		Limit: 0,
+	})
```

Comment:
same.

either bucketDB should not have a pagination interface, or it should and we should use it.